crates/asm/common/src/aux.rs

```diff
@@ -1,56 +1,87 @@
 
-/// A single subprotocol's auxiliary input payload, containing processed auxiliary data.
-///
-/// Each [`AuxRequest`] must resolve into a corresponding [`AuxPayload`] before the main
-/// processing phase can begin. The `data` field must deserialize into an instance of
-/// [`Subprotocol::AuxInput`] for the associated subprotocol.
-#[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
-pub struct AuxPayload {
-    /// Processed auxiliary input data as raw bytes.
-    ///
-    /// This `Vec<u8>` must deserialize into one
-    /// `<P as Subprotocol>::AuxInput` for the corresponding subprotocol P.
-    pub data: Vec<u8>,
+/// Per-block set of historical logs together with an MMR inclusion proof.
+#[derive(Debug, Clone)]
+pub struct BlockLogsOracleData {
+    pub block_hash: BlockHash,
+    pub logs: Vec<AsmLogEntry>,
+    pub proof: LogMmrProof,
 }
 
-impl AuxPayload {
-    pub fn new(data: Vec<u8>) -> Self {
-        Self { data }
+impl BorshSerialize for BlockLogsOracleData {
+    fn serialize<W: Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        self.block_hash.to_byte_array().serialize(writer)?;
+        self.logs.serialize(writer)?;
+        let cohashes: Vec<[u8; 32]> = self.proof.cohashes().to_vec();
+        cohashes.serialize(writer)?;
+        self.proof.index().serialize(writer)?;
+        Ok(())
     }
+}
 
-    pub fn data(&self) -> &[u8] {
-        &self.data
+impl BorshDeserialize for BlockLogsOracleData {
+    fn deserialize_reader<R: Read>(reader: &mut R) -> Result<Self, std::io::Error> {
+        let block_hash_bytes = <[u8; 32]>::deserialize_reader(reader)?;
+        let logs = Vec::<AsmLogEntry>::deserialize_reader(reader)?;
+        let cohashes = Vec::<[u8; 32]>::deserialize_reader(reader)?;
+        let index = u64::deserialize_reader(reader)?;
+        let proof = LogMmrProof::from_cohashes(cohashes, index);
+        Ok(Self {
+            block_hash: BlockHash::from_byte_array(block_hash_bytes),
+            logs,
+            proof,
+        })
     }
+}
 
-    /// Tries to parse as a subprotocol's aux input.
-    ///
-    /// This MUST NOT be called on a payload that does not correspond to the
-    /// subprotocol type, because this may lead to silent errors.
-    pub fn try_to_aux_input<S: Subprotocol>(&self) -> Result<S::AuxInput, AsmError> {
-        <S::AuxInput as BorshDeserialize>::try_from_slice(&self.data)
-            .map_err(|e| AsmError::Deserialization(S::ID, e))
-    }
+/// DRT transaction auxiliary data response.
+// TODO: Update field type as needed.
+#[derive(Debug, Clone, BorshDeserialize, BorshSerialize)]
+pub struct DrtTxOracleData {
+    pub drt_tx: Vec<u8>,
+}
+
+#[derive(Debug, Clone, BorshDeserialize, BorshSerialize)]
+pub struct AuxData {
+    pub asm_log_oracles: Vec<BlockLogsOracleData>,
+    pub drt_tx_oracles: Vec<DrtTxOracleData>,
+}
+
+#[derive(Debug, Default, Clone, BorshDeserialize, BorshSerialize)]
+pub struct AuxInput {
+    pub data: BTreeMap<L1TxIndex, AuxData>,
 }
```

Comment:
How would we pass aux data if there wasn't one particular tx that the request pertained to?  For example, if it was aggregated from multiple txs or the requests were from on from data in the subprotos state.

Really we would just want to pass the `AuxData` as the whole struct and have the subproto impl figure out what it needs from that instead of enforcing a structure binding the data entries to a tx.